pkg/controller/seed-controller-manager/etcdbackup/etcd_backup_controller.go

```diff
@@ -0,0 +1,983 @@
+		// we can just delete the backup from backupConfig.Status.CurrentBackups directly
+		if backupConfig.DeletionTimestamp != nil && backup.BackupPhase == "" {
+			continue
+		}
+
+		backupJobDeleted := false
+		if backup.BackupFinishedTime != nil {
+			var retentionTime time.Duration
+			switch {
+			case backupConfig.DeletionTimestamp != nil:
+				retentionTime = 0
+			case backup.BackupPhase == kubermaticv1.BackupStatusPhaseCompleted:
+				retentionTime = succeededJobRetentionTime
+			default:
+				retentionTime = failedJobRetentionTime
+
+			}
+
+			age := r.clock.Now().Sub(backup.BackupFinishedTime.Time)
+
+			if age < retentionTime {
+				// don't delete the job yet, but reconcile when the time has come to delete it
+				returnReconcile = minReconcile(returnReconcile, &reconcile.Result{RequeueAfter: retentionTime - age})
+
+			} else {
+				// delete job
+				job := &batchv1.Job{}
+
+				err := r.Get(ctx, types.NamespacedName{Namespace: metav1.NamespaceSystem, Name: backup.JobName}, job)
+				switch {
+				case kerrors.IsNotFound(err):
+					backupJobDeleted = true
+				case err == nil:
+					err := r.Delete(ctx, job, ctrlruntimeclient.PropagationPolicy(metav1.DeletePropagationBackground))
+					if err != nil && !kerrors.IsNotFound(err) {
+						return nil, errors.Wrapf(err, "backup %s: failed to delete backup job %s", backup.BackupName, backup.JobName)
+					}
+					backupJobDeleted = true
+				case !kerrors.IsNotFound(err):
+					return nil, errors.Wrapf(err, "backup %s: failed to get backup job %s", backup.BackupName, backup.JobName)
+				}
+
+			}
+		}
+
+		deleteJobDeleted := false
+		if backup.DeleteFinishedTime != nil {
+			var retentionTime time.Duration
+			if backupConfig.DeletionTimestamp != nil {
+				retentionTime = 0
+			} else {
+				retentionTime = succeededJobRetentionTime
+			}
+
+			age := r.clock.Now().Sub(backup.DeleteFinishedTime.Time)
+
+			if age < retentionTime {
+				// don't delete the job yet, but reconcile when the time has come to delete it
+				returnReconcile = minReconcile(returnReconcile, &reconcile.Result{RequeueAfter: retentionTime - age})
+
+			} else {
+				// delete job
+				job := &batchv1.Job{}
+
+				err := r.Get(ctx, types.NamespacedName{Namespace: metav1.NamespaceSystem, Name: backup.DeleteJobName}, job)
+				switch {
+				case kerrors.IsNotFound(err):
+					deleteJobDeleted = true
+				case err == nil:
+					err := r.Delete(ctx, job, ctrlruntimeclient.PropagationPolicy(metav1.DeletePropagationBackground))
+					if err != nil && !kerrors.IsNotFound(err) {
+						return nil, errors.Wrapf(err, "backup %s: failed to delete job %s", backup.BackupName, backup.DeleteJobName)
+					}
+					deleteJobDeleted = true
+				case !kerrors.IsNotFound(err):
+					return nil, errors.Wrapf(err, "backup %s: failed to get job %s", backup.BackupName, backup.DeleteJobName)
+				}
+			}
+		}
+
+		if backupJobDeleted && deleteJobDeleted {
+			continue // delete backup from backupConfig.Status.CurrentBackups
```

Comment:
I'm skipping it in newBackups, but since I'm writing that to backupConfig.Status.CurrentBackups in the end, I'll end up deleting it from there. I'll try to make that more explicit in the comment.